sc-memory/tests/sc-memory/common/test_sc_elements.cpp

```diff
@@ -299,6 +299,66 @@ TEST_F(ScMemoryTest, CreateEdges)
   EXPECT_TRUE(ctx.HelperCheckEdge(linkAddr, nodeAddr, ScType::EdgeUCommonConst));
 }
 
+TEST_F(ScMemoryTest, IterateEraseEdgesBetweenNodes)
+{
+  ScAddr const classAddr = m_ctx->CreateNode(ScType::NodeConstClass);
+  ScAddr const nodeAddr = m_ctx->CreateNode(ScType::NodeConst);
+  
+  size_t const edgesCount = 5;
+  for (size_t i = 0; i < edgesCount; ++i)
+    m_ctx->CreateEdge(ScType::EdgeAccessConstPosPerm, classAddr, nodeAddr);
+
+  size_t i = 0;
+  ScIterator3Ptr const it = m_ctx->Iterator3(classAddr, ScType::EdgeAccessConstPosPerm, nodeAddr);
+  while (it->Next())
+  {
+    m_ctx->EraseElement(it->Get(1));
+    ++i;
+  }
+
+  EXPECT_EQ(i, edgesCount);
+}
+
+TEST_F(ScMemoryTest, IterateEraseEdgesFromNode)
+{
+  ScAddr const classAddr = m_ctx->CreateNode(ScType::NodeConstClass);
+  ScAddr const nodeAddr = m_ctx->CreateNode(ScType::NodeConst);
+  
+  size_t const edgesCount = 5;
+  for (size_t i = 0; i < edgesCount; ++i)
+    m_ctx->CreateEdge(ScType::EdgeAccessConstPosPerm, classAddr, nodeAddr);
+
+  size_t i = 0;
+  ScIterator3Ptr const it = m_ctx->Iterator3(classAddr, ScType::EdgeAccessConstPosPerm, ScType::NodeConst);
+  while (it->Next())
+  {
+    m_ctx->EraseElement(it->Get(1));
+    ++i;
+  }
+
+  EXPECT_EQ(i, edgesCount);
+}
+
+TEST_F(ScMemoryTest, IterateEraseEdgesToNode)
+{
+  ScAddr const classAddr = m_ctx->CreateNode(ScType::NodeConstClass);
+  ScAddr const nodeAddr = m_ctx->CreateNode(ScType::NodeConst);
+  
+  size_t const edgesCount = 5;
+  for (size_t i = 0; i < edgesCount; ++i)
+    m_ctx->CreateEdge(ScType::EdgeAccessConstPosPerm, classAddr, nodeAddr);
+
+  size_t i = 0;
+  ScIterator3Ptr const it = m_ctx->Iterator3(ScType::NodeConstClass, ScType::EdgeAccessConstPosPerm, nodeAddr);
+  while (it->Next())
+  {
+    m_ctx->EraseElement(it->Get(1));
+    ++i;
+  }
+
+  EXPECT_EQ(i, edgesCount);
+}
+
```

Comment:
also can you check what will happen if 2 edges are removed using first iterator, and then rest of edges are removed using second iterator